app/bundles/LeadBundle/Helper/AnonymizeHelper.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\LeadBundle\Helper;
+
+class AnonymizeHelper
+{
+    public const PRE_DEFINED_DOMAIN       = 'ano.nym';
+    public const PRE_PSEUDONYMIZED_DOMAIN = 'pseudo.nym';
+
+    /**
+     * @param string|bool|null $email
+     */
+    public static function email($email, bool $pseudonymized = false, string $newDomain = self::PRE_DEFINED_DOMAIN): string|false
+    {
+        if (empty($email)) {
+            return '';
+        }
+
+        if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+            return false;
+        }
```

Comment:
```suggestion
    public static function email($email, bool $pseudonymized = false, string $newDomain = self::PRE_DEFINED_DOMAIN): ?string
    {
        if (empty($email) || !filter_var($email, FILTER_VALIDATE_EMAIL)) {
            return null;
        }
```

Lets avoid mixed return types (string, false, empty string).